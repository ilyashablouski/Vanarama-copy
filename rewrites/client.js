/* eslint-disable no-console */
const fetch = require('node-fetch');

const {
  ApolloClient,
  createHttpLink,
  InMemoryCache,
} = require('@apollo/client');

// const inspect = require('../inspect');

const httpLink = createHttpLink({
  uri: process.env.API_URL,
  fetch,
  headers: {
    'x-api-key': process.env.API_KEY,
  },
});

const client = new ApolloClient({
  link: httpLink,
  cache: new InMemoryCache(),
  onError: ({ networkError, graphQLErrors }) => {
    console.log('graphQLErrors', graphQLErrors);
    console.log('networkError', networkError);
  },
});

module.exports = client;
