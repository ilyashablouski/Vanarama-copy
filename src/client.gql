extend type Query {
  selectedOrderUuid: ID
  selectedDerivativeId: ID
}
