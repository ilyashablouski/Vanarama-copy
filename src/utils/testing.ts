// eslint-disable-next-line import/no-extraneous-dependencies
import { fireEvent } from '@testing-library/react';

// eslint-disable-next-line import/prefer-default-export
export const inputChange = (element: Element, value: string) => {
  fireEvent.input(element, {
    target: { value },
  });
};
