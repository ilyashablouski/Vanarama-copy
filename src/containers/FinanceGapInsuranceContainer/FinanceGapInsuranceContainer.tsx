import dynamic from 'next/dynamic';
import { useState } from 'react';
import * as toast from 'core/atoms/toast/Toast';
import Router from 'next/router';
import Breadcrumb from 'core/atoms/breadcrumb-v2';
import GoldrushForm from '../../components/GoldrushForm/GoldrushForm';
import { IGoldrushFromValues } from '../../components/GoldrushForm/interfaces';
import InsuranceHeroSection from '../InsurancePageContainer/sections/InsuranceHeroSection';
import { GenericPageQuery_genericPage_sections as Section } from '../../../generated/GenericPageQuery';
import InsuranceTypeSection from './sections/InsuranceTypeSection';
import InsuranceFormSection from './sections/InsuranceFormSection';
import { useOpportunityCreation } from '../GoldrushFormContainer/gql';
import {
  OpportunitySubtypeEnum,
  OpportunityTypeEnum,
} from '../../../generated/globalTypes';
import { pushInsuranceEventDataLayer } from '../../utils/dataLayerHelpers';

const Modal = dynamic(() => import('core/molecules/modal'));

interface IProps {
  sections: Section | null | undefined;
  breadcrumbsData: any;
}

export const handleNetworkError = () =>
  toast.error(
    'Sorry there seems to be an issue with your request. Pleaser try again in a few moments',
    '',
  );

const toThankYouPage = () => {
  /*
   *  TODO: van-insurance/ should be replaced with insurance/
   *   in the redirect url when we completely stop using van-insurance pages.
   */
  const asPath = Router.asPath.match(/([\w-]+)(\.html)?$/i)?.[1];
  return Router.push(`/van-insurance/${asPath}/thank-you`);
};

const FinanceGapInsurancePageContainer = ({
  sections,
  breadcrumbsData,
}: IProps) => {
  const hero = sections?.hero;
  const leadText = sections?.leadText;
  const featured1 = sections?.featured1;
  const featured2 = sections?.featured2;
  const rowText = sections?.rowText;

  const [showModal, setShowModal] = useState(false);

  const [createOpportunity, { loading }] = useOpportunityCreation(
    () => toThankYouPage(),
    error => {
      if (error?.networkError) {
        handleNetworkError();
      }
    },
  );

  const breadcrumbsItems = breadcrumbsData?.map((el: any) => ({
    link: { href: el.href || '', label: el.label },
  }));

  const getOpportunitySubtype = () => {
    switch (Router.asPath) {
      case 'insurance/short-term-insurance':
      case '/van-insurance/short-term-insurance.html':
        return OpportunitySubtypeEnum.SHORTTERM;
      case 'insurance/tools-in-transit':
      case '/van-insurance/tools-in-transit.html':
        return OpportunitySubtypeEnum.TOOLSINTRANSIT;
      case 'insurance/finance-gap-insurance':
      case '/van-insurance/finance-gap-insurance.html':
        return OpportunitySubtypeEnum.GAPINSURANCE;
      case 'insurance/multi-year-insurance':
      case '/van-insurance/multi-year-van-insurance.html':
        return OpportunitySubtypeEnum.MULTIYEAR;
      default:
        return null;
    }
  };

  return (
    <>
      {hero && <InsuranceHeroSection {...hero} />}
      {breadcrumbsItems && (
        <div className="row:title -mt-200">
          <Breadcrumb items={breadcrumbsItems} />
        </div>
      )}
      {leadText && <InsuranceTypeSection {...leadText} />}
      {featured1 && (
        <InsuranceFormSection
          {...featured1}
          isSubmitting={loading}
          onSubmit={(values: IGoldrushFromValues) => {
            createOpportunity({
              variables: {
                email: values.email,
                phoneNumber: values.phoneNumber,
                fullName: values.fullName,
                postcode: values.postcode,
                opportunityType: OpportunityTypeEnum.INSURANCE,
                termsAndConditions: Boolean(values.termsAndCons),
                privacyPolicy: Boolean(values.privacyPolicy),
                communicationsConsent: Boolean(values.consent),
                opportunitySubtype: getOpportunitySubtype(),
              },
            }).then(() =>
              setTimeout(() => {
                pushInsuranceEventDataLayer(Router);
              }, 200),
            );
          }}
        />
      )}
      {showModal && (
        <Modal
          className="-mt-000 callBack"
          show
          onRequestClose={() => {
            setShowModal(false);
          }}
        >
          <GoldrushForm
            callBack={false}
            className="form-insurance"
            isSubmitting={loading}
            isPostcodeVisible
            termsAndConditionsId="modal"
            isTextInVisible
            noTermsAndConditions
            onSubmit={(values: IGoldrushFromValues) => {
              createOpportunity({
                variables: {
                  email: values.email,
                  phoneNumber: values.phoneNumber,
                  fullName: values.fullName,
                  postcode: values.postcode,
                  opportunityType: OpportunityTypeEnum.INSURANCE,
                  termsAndConditions: Boolean(values.termsAndCons),
                  privacyPolicy: Boolean(values.privacyPolicy),
                  communicationsConsent: Boolean(values.consent),
                  opportunitySubtype: getOpportunitySubtype(),
                },
              }).then(() =>
                setTimeout(() => {
                  pushInsuranceEventDataLayer(Router);
                }, 200),
              );
            }}
          />
        </Modal>
      )}
      <hr className="-fullwidth" />
      {rowText && (
        <InsuranceTypeSection
          showModal={() => setShowModal(true)}
          {...rowText}
          link1={featured1?.link}
          link2={featured2?.link}
        />
      )}
    </>
  );
};

export default FinanceGapInsurancePageContainer;
