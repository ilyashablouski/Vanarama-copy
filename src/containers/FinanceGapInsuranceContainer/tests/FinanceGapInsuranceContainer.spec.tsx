import React from 'react';
// @ts-ignore
import preloadAll from 'jest-next-dynamic';
import { MockedProvider } from '@apollo/client/testing';
import { screen, render, waitFor } from '@testing-library/react';
import FinanceGapInsuranceContainer from '../FinanceGapInsuranceContainer';

// ARRANGE
const SECTIONS = {
  leadText: {
    titleTag: null,
    heading: 'Guaranteed Asset Protection (GAP)',
    description:
      'Cover the difference between your insurance payment and the full value of your car.',
    position: 1,
  },
  featured: null,
  rowText: {
    position: 3,
    heading: 'Get A Quote',
    titleTag: null,
    subHeading: null,
  },
  featured1: {
    title: 'Finance Gap Explained',
    body:
      'Vehicle value depreciation could mean there is a risk you will owe more to the Finance Company than the Insurance Company pay-out if the vehicle is written off or stolen. Finance Gap Insurance allows you to insure against the difference (up to a maximum of £20,000) between your Insurance Company pay-out and the amount you owe to the Finance Company. Check out the video below for more information:\n\n<a href="https://www.youtube.com/embed/P4IT21lvn4w" class="embedly-card" data-card-width="100%" data-card-controls="0">Embedded content: https://www.youtube.com/embed/P4IT21lvn4w</a>\n\nFull terms and conditions can be found here.\n\n## Eligible Vehicles\n- Vanarama leased vehicles only\n- Vehicles up to £125,000 Invoice price\n- Cars & Commercial vehicles up to 3.5cwt\n- Cover is valid for up to 5 years in line with the finance agreement\n- Applies to all finance agreements, including - Contract Hire and Lease\n- Applies to cars and commercial vehicles up to 6 years old\n\n## Major Exclusions\n- If the vehicle is written off whilst driving illegally or under the influence of drugs or alcohol\n- The vehicle is not covered by a Fully Comprehensive Road Risk Motor Policy\n- Use of vehicle for Hire or Reward\n- If the vehicle is unattended and the keys have NOT been removed, all doors locked and any immobilizer and/or alarm activated\n- If your vehicle has a Fridge Unit/Conversion & also if your vehicle has/will have a Motor Traders Insurance Policy\n- Once an order has been placed, a member of our dedicated GAP team will be in contact to discuss your options\n- The vehicle must be separately insured by a Fully Comprehensive Road Risk Motor Policy\n\nIf you would like to know more, call our dedicated Finance Gap Insurance team now on 01442 835768.',
    titleTag: null,
    link: {
      url: 'https://beta.vanarama.com/van-insurance.html',
      text: 'Get A Quote',
    },
    iconList: null,
  },
  featured2: {
    title: null,
    body: null,
    titleTag: null,
    link: {
      url: 'https://beta.vanarama.com/contact-us.html',
      text: 'Get In Touch',
    },
  },
  faqs: null,
  hero: {
    position: 0,
    flag: null,
    title: null,
    titleTag: null,
    body: '## Extra Protection Should The Worst happen',
    image: null,
    heroCard: [
      {
        title: 'Contact Us',
        body:
          '\n|      |      |\n| ---------- | ---------- |\n| Monday–Thursday | 8:30am–7pm |\n| Friday | 9am–5pm |\n| Saturday | 9am–3pm |\n| Sunday | Closed |\n\n[Call 01442 838173]("Call 01442 838173")\n\n',
      },
    ],
  },
  tiles: null,
  cards: null,
} as any;

describe('<FinanceExplainedContainer />', () => {
  beforeAll(async () => {
    await preloadAll();
  });

  it('should match snapshot', async () => {
    // ACT
    const getComponent = render(
      <MockedProvider addTypename={false}>
        <FinanceGapInsuranceContainer sections={SECTIONS} />
      </MockedProvider>,
    );
    // ASSERT
    await waitFor(() => {
      expect(screen.getByText(`Finance Gap Explained`)).toBeInTheDocument();
    });

    await waitFor(() => {
      expect(
        screen.getByText(`Guaranteed Asset Protection (GAP)`),
      ).toBeInTheDocument();
    });

    await waitFor(() => {
      expect(screen.getByText(`Get In Touch`)).toBeInTheDocument();
    });

    const tree = getComponent.baseElement;
    expect(tree).toMatchSnapshot();
  });
});
