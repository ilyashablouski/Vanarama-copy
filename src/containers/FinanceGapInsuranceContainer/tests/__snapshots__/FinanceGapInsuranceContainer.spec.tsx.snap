// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FinanceExplainedContainer /> should match snapshot 1`] = `
<body>
  <div>
    <div
      class="row:bg-hero"
    >
      <div
        class="row:hero"
      >
        <div
          class="hero--left"
        >
          <p
            class="heading -white -xlarge"
          />
          <p
            class="text -large -white"
          >
            Extra Protection Should The Worst happen
          </p>
          <div
            class="image hero--image -expand -plain"
          >
            <img
              alt="Hero Image"
              class="image--native"
              src="https://www.vanarama.com/assets/images/insurance/home-hero/insurance-hero.png"
            />
          </div>
        </div>
        <div
          class="hero--right"
        >
          <div
            class="card hero-card"
            data-testid="card"
          >
            <div
              class="hero-card--inner"
            >
              <p
                class="heading -black -lead"
              >
                Contact Us
              </p>
              <section
                class="structured-list -styled-headers -small -opening-times"
              >
                <div
                  class="structured-list-tbody"
                >
                  <div
                    class="structured-list-row "
                  >
                    <div
                      class="structured-list-td structured-list-content--nowrap"
                      data-testid="data_id-0-Monday–Thursday"
                    >
                      Monday–Thursday
                    </div>
                    <div
                      class="structured-list-td"
                    >
                      8:30am–7pm
                    </div>
                  </div>
                  <div
                    class="structured-list-row "
                  >
                    <div
                      class="structured-list-td structured-list-content--nowrap"
                      data-testid="data_id-1-Friday"
                    >
                      Friday
                    </div>
                    <div
                      class="structured-list-td"
                    >
                      9am–5pm
                    </div>
                  </div>
                  <div
                    class="structured-list-row "
                  >
                    <div
                      class="structured-list-td structured-list-content--nowrap"
                      data-testid="data_id-2-Saturday"
                    >
                      Saturday
                    </div>
                    <div
                      class="structured-list-td"
                    >
                      9am–3pm
                    </div>
                  </div>
                  <div
                    class="structured-list-row "
                  >
                    <div
                      class="structured-list-td structured-list-content--nowrap"
                      data-testid="data_id-3-Sunday"
                    >
                      Sunday
                    </div>
                    <div
                      class="structured-list-td"
                    >
                      Closed
                    </div>
                  </div>
                </div>
              </section>
              <a
                class="link -primary -regular -plain"
                href="tel:01442838173"
              >
                <button
                  class="button -fullwidth -teal -large -solid"
                >
                  <div
                    class="button--inner"
                  >
                    Call 01442 838173
                  </div>
                </button>
              </a>
            </div>
          </div>
        </div>
        <div
          class="hero--decals"
        >
          <svg
            aria-hidden="true"
            class="hero-logomark"
            focusable="false"
            height="318"
            viewBox="0 0 320 318"
            width="320"
          >
            <defs>
              <clippath
                id="clip-Vanarama_Logo"
              >
                <rect
                  height="318"
                  width="320"
                />
              </clippath>
            </defs>
            <g
              clip-path="url(#clip-Vanarama_Logo)"
              data-name="Vanarama Logo"
              id="Vanarama_Logo"
            >
              <g
                data-name="Group 1"
                id="Group_1"
                transform="translate(-548.96 -430.411)"
              >
                <path
                  d="M791.259,591.381a17.785,17.785,0,0,0-17.84-17.729c-.222,0-.426.058-.645.067a17.72,17.72,0,0,0-9.362-32.84h0s-58.89.05-63.18,0c0,0,14.286-38.444,17.34-46.827a64.077,64.077,0,0,0,3.366-12.624c2.018-14.048-4.7-27.255-16.6-33.271-.3.472-49.532,89.221-67.931,120.809l.165-.067c-7.728,13.919-10.778,29.466-8.928,46.881,2.982,28.091,20.387,48.953,47.761,55.979,6.445,1.654,13.126,2.388,19.7,3.545l60.518-.194h0l.72,0h0l.667,0,.007-.08a17.7,17.7,0,0,0,8.4-32.663c.143.006.272.046.416.048l.717,0a17.706,17.706,0,0,0,8.032-33.307A17.726,17.726,0,0,0,791.259,591.381Z"
                  data-name="Path 1"
                  id="Path_1"
                />
                <path
                  d="M724.251,431.411a55.254,55.254,0,0,1,18.839,38.628,124.188,124.188,0,1,1-81.74,4.495c7.328-13.124,23.034-41.338,23.489-42.1C608.592,443.958,549.96,509.924,549.96,589.337c0,87.508,71.2,158.7,158.7,158.7s158.7-71.2,158.7-158.7C867.362,507.088,804.469,439.27,724.251,431.411Z"
                  data-name="Path 2"
                  id="Path_2"
                />
              </g>
            </g>
          </svg>
          <svg
            class="hero--curve"
            data-name="curve"
            id="hero--curve"
            viewBox="0 0 1920 170.2"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="hero--curve-background"
              d="M0,65.16v107H1920V113.35s-271.62,32.89-625.91,42.9C925.77,179.08,394.27,183,0,65.16Z"
              transform="translate(0 -2)"
            />
            <path
              class="hero--curve-foreground"
              d="M0,2V64.75c394.27,117.82,925.77,113.92,1294.09,91.08C874,167.71,337.57,147.42,0,2Z"
              transform="translate(0 -1)"
            />
          </svg>
          <svg
            class="hero--curve-m"
            data-name="curve"
            id="hero--curve-m"
            viewBox="0 0 375 65.05"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="hero--curve-foreground"
              d="M0,1.81v35c122.83,16.41,249.92,25.7,375,30V47.6C246.33,38.57,119.07,24,0,1.81Z"
              transform="translate(0 -1)"
            />
            <path
              class="hero--curve-background"
              d="M0,66.86H375c-125.08-4.32-252.17-13.61-375-30Z"
              transform="translate(0 -1.81)"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="row:lead-text"
    >
      <p
        class="heading -black -xlarge"
      >
        Guaranteed Asset Protection (GAP)
      </p>
      <p
        class="text -lead -darker"
      >
        Cover the difference between your insurance payment and the full value of your car.
      </p>
    </div>
    <div
      class="row:featured-right"
    >
      <div>
        <p
          class="heading -black -large"
        >
          Finance Gap Explained
        </p>
        <p>
          Vehicle value depreciation could mean there is a risk you will owe more to the Finance Company than the Insurance Company pay-out if the vehicle is written off or stolen. Finance Gap Insurance allows you to insure against the difference (up to a maximum of £20,000) between your Insurance Company pay-out and the amount you owe to the Finance Company. Check out the video below for more information:
        </p>
        <p>
          <span>
            <a
              class="embedly-card"
              data-card-controls="0"
              data-card-width="100%"
              href="https://www.youtube.com/embed/P4IT21lvn4w"
            />
          </span>
          Embedded content: https://www.youtube.com/embed/P4IT21lvn4w
          <span />
        </p>
        <p>
          Full terms and conditions can be found here.
        </p>
        <h2>
          Eligible Vehicles
        </h2>
        <ul>
          <li>
            Vanarama leased vehicles only
          </li>
          <li>
            Vehicles up to £125,000 Invoice price
          </li>
          <li>
            Cars & Commercial vehicles up to 3.5cwt
          </li>
          <li>
            Cover is valid for up to 5 years in line with the finance agreement
          </li>
          <li>
            Applies to all finance agreements, including - Contract Hire and Lease
          </li>
          <li>
            Applies to cars and commercial vehicles up to 6 years old
          </li>
        </ul>
        <h2>
          Major Exclusions
        </h2>
        <ul>
          <li>
            If the vehicle is written off whilst driving illegally or under the influence of drugs or alcohol
          </li>
          <li>
            The vehicle is not covered by a Fully Comprehensive Road Risk Motor Policy
          </li>
          <li>
            Use of vehicle for Hire or Reward
          </li>
          <li>
            If the vehicle is unattended and the keys have NOT been removed, all doors locked and any immobilizer and/or alarm activated
          </li>
          <li>
            If your vehicle has a Fridge Unit/Conversion & also if your vehicle has/will have a Motor Traders Insurance Policy
          </li>
          <li>
            Once an order has been placed, a member of our dedicated GAP team will be in contact to discuss your options
          </li>
          <li>
            The vehicle must be separately insured by a Fully Comprehensive Road Risk Motor Policy
          </li>
        </ul>
        <p>
          If you would like to know more, call our dedicated Finance Gap Insurance team now on 01442 835768.
        </p>
      </div>
      <div
        class="card"
      >
        <form
          class="form"
        >
          <div
            class="formgroup"
          >
            <label
              class="formgroup--legend"
              for="goldrush-form_full-name"
            >
              Full Name
            </label>
            <div
              class="formgroup--content"
            >
              <div
                class="textinput regular"
              >
                <input
                  class="textinput--native regular"
                  data-testid="goldrush-form_full-name"
                  id="goldrush-form_full-name"
                  name="fullName"
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            class="formgroup"
          >
            <label
              class="formgroup--legend"
              for="goldrush-form_email"
            >
              Email Address
            </label>
            <div
              class="formgroup--content"
            >
              <div
                class="textinput regular"
              >
                <input
                  class="textinput--native regular"
                  data-testid="goldrush-form_email"
                  id="goldrush-form_email"
                  name="email"
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            class="formgroup"
          >
            <label
              class="formgroup--legend"
              for="goldrush-form_phone-number"
            >
              Phone Number
            </label>
            <div
              class="formgroup--content"
            >
              <div
                class="textinput regular"
              >
                <input
                  class="textinput--native regular"
                  data-testid="goldrush-form_phone-number"
                  id="goldrush-form_phone-number"
                  name="phoneNumber"
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            class="formgroup"
          >
            <label
              class="formgroup--legend"
              for="goldrush-form_postcode"
            >
              Postcode
            </label>
            <div
              class="formgroup--content"
            >
              <div
                class="textinput regular"
              >
                <input
                  class="textinput--native regular"
                  data-testid="goldrush-form_postcode"
                  id="goldrush-form_postcode"
                  name="postcode"
                  type="text"
                />
              </div>
            </div>
          </div>
          <span
            class="text -small -darker"
          >
            Terms and conditions agreement text and
             
            <a
              class="link -primary -regular -plain"
              href=""
            >
              link
            </a>
          </span>
          <button
            class="button -teal -regular -solid"
          >
            <div
              class="button--inner"
            >
              Get Quote Now
            </div>
          </button>
        </form>
      </div>
    </div>
    <hr
      class="-fullwidth"
    />
    <div
      class="row:lead-text"
    >
      <p
        class="heading -black -xlarge"
      >
        Get A Quote
      </p>
      <div
        class="button-group"
      >
        <button
          class="button -teal -regular -solid"
        >
          <div
            class="button--inner"
          >
            Get A Quote
          </div>
        </button>
        <button
          class="button -teal -regular -outline"
        >
          <div
            class="button--inner"
          >
            Get In Touch
          </div>
        </button>
      </div>
    </div>
  </div>
</body>
`;
